pkg/operator/controller/ingress/deployment.go

```diff
@@ -559,8 +569,20 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 
 	nodeSelector := map[string]string{
 		"kubernetes.io/os":               "linux",
-		"node-role.kubernetes.io/worker": "",
 	}
+
+	switch ingressConfig.Status.DefaultPlacement {
+	case configv1.DefaultPlacementControlPlane:
+		nodeSelector["node-role.kubernetes.io/master"] = ""
+	default:
+		nodeSelector["node-role.kubernetes.io/worker"] = ""
+	}
+
+	if ingressConfig.Status.DefaultPlacement == configv1.DefaultPlacementControlPlane {
+		delete(nodeSelector, "node-role.kubernetes.io/worker")
```

Comment:
The `delete` can be deleted.

Reply:
Actually, this whole if statement can be.

Reply:
Ah yeah obviously